### DISCUSSION TITLE
VSCode Web Panel won't respect the CSP header

### DISCUSSION THREAD

**Discussion by jakereifer** [microsoft/vscode-discussions#949] [Extension Development QnA] [answered]
So I have a web panel that has a simple form which sends data to 'https://localhost:3456/'. Within the html, i have a `meta` tag that adds 

`<meta http-equiv="Content-Security-Policy" content="default-src 'none'; frame-src ${fullWebServerUri} ${cspSource}; ... /meta>`

where `fullWebServerUri` = `'https://localhost:3456/'` and `cspSource` = `webview.panel.cspSource`.

When i load the webview panel and submit the form, I dont get the data on localhost:3456 and using the dev tools I see that it didn't go through because:

`Refused to frame 'https://localhost:3456/' because it violates the following Content Security Policy directive: "frame-src 'self'".`

It seems to be ignoring my csp header (which i followed the [docs](https://code.visualstudio.com/api/extension-guides/webview#content-security-policy)). Any ideas? Is there a more general CSP i need to override?

**Comment by tamuratak** [comment, marked as answer]
It is forbidden to redirect the WebView to `https://localhost:3456/`, or any other url. The form action redirects it to there as you know. The content of the HTML at `https://localhost:3456/` has nothing to do with it. The parent frame of the WebView forbids it.

See `frame-src 'self'` at https://github.com/microsoft/vscode/blob/6aa2a71ddd097aaa4930e54a908812a0a77d34e8/src/vs/workbench/contrib/webview/browser/pre/index.html#L8, which forbids the redirect. You can use the `Developer: Toggle Developer Tools` command to check it.

>         content="default-src 'none'; script-src 'sha256-frEVWVmmI4TWHGHXZaCTWqGQI9jv+i8hv+sOa87Gqlc=' 'self'; frame-src 'self'; style-src 'unsafe-inline';">

An easy workaround is making an iframe whose src is `https://localhost:3456/` in the WebView.
